PyInstaller/utils/win32/winmanifest.py

```diff
@@ -1072,7 +1072,9 @@ def create_manifest(filename, manifest, console, uac_admin=False, uac_uiaccess=F
     # only write a new manifest if it is different from the old
     need_new = not os.path.exists(filename)
     if not need_new:
-        with open(filename) as f:
+        # WARNING: Don't rely on the default encoding of "open" function.
+        # It can be not allowed one for XML. (2018-04-23, Python 3.6)
+        with open(filename, encoding="UTF-8") as f:
```

Comment:
Unfortunately the `encoding` parameter is not supported in Python2.7, see https://docs.python.org/2/library/functions.html#open.

Also I wonder: This assumes the XML to be UTF-8, which might not be the case.

I suggest changing this part (which was introduced in d41d31a9f7d930876492ace6) to use `ManifestFromXMLFile(filename)`, which uses `xml.dom.minidom.parse()` and thus should encapsulate the proper decoding.

Reply:
OK, That's good idea to use `ManifestFromXMLFile()`. I updated the pull-request.
`Manifest` object always uses `utf-8` as default, that's free from the behavior of built-in `open` function.